packages/react-form/src/hooks/submit.ts

```diff
@@ -1,40 +1,49 @@
-import {useState, useCallback, useRef} from 'react';
-import {useMountedRef} from '@shopify/react-hooks';
+import {useState, useCallback} from 'react';
+import {useLazyRef, useMountedRef} from '@shopify/react-hooks';
 
 import {
   FormMapping,
   SubmitHandler,
   SubmitResult,
   FieldBag,
   FormError,
+  DynamicListBag,
 } from '../types';
 import {
   propagateErrors,
   validateAll,
   getValues,
   makeCleanFields,
+  makeCleanDynamicList,
 } from '../utilities';
 
 export function useSubmit<T extends FieldBag>(
   onSubmit: SubmitHandler<FormMapping<T, 'value'>> = noopSubmission,
   fieldBag: T,
   makeCleanAfterSubmit = false,
+  dynamicListBag?: DynamicListBag,
 ) {
   const mounted = useMountedRef();
   const [submitting, setSubmitting] = useState(false);
   const [submitErrors, setSubmitErrors] = useState([] as FormError[]);
 
-  const fieldBagRef = useRef(fieldBag);
+  const fieldBagRef = useLazyRef(() => fieldBag);
   fieldBagRef.current = fieldBag;
+  const dynamicListBagRef = useLazyRef(() => dynamicListBag);
+  dynamicListBagRef.current = dynamicListBag;
 
-  const setErrors = useCallback((errors: FormError[]) => {
-    setSubmitErrors(errors);
-    propagateErrors(fieldBagRef.current, errors);
-  }, []);
+  const setErrors = useCallback(
+    (errors: FormError[]) => {
+      setSubmitErrors(errors);
+      propagateErrors(fieldBagRef.current, errors);
```

Comment:
Seeing this makes me feel the error handling might be incomplete when passing `dynamicLists` 👀

Reply:
🤔 It should not be incomplete because the dynamic lists fields are in the fieldBag. Any errors in the fields should show up here too :

Reply:
OOps this comment should had been removed, all good!